### DISCUSSION TITLE
由农场的APY算法实例，分析影响因素关系。

### DISCUSSION THREAD

**Discussion by yancanlou** [famland/website#97] [Show and tell]
一个农场APY估算如下：
其农场基本数据：
**分发池额为：27091yl，共享值总量：15849U，田：30块，（田均投入15849U/30块=528.3U），yl单价（0.49U）**
可获收益：K=27091yl x 30%=8127yl；
平均最小本金田(无原始田)：M最小情况=15849U/30块/yl单价（0.49U）x30倍年化=32344yl,
平均最大本金田(有原始田)：M最大情况=15849U/30块/yl单价（0.49U）x30倍年化+1000U/yl单价（0.49U）x30倍年化；=(15849/30+1000)/0.49x30=93569;
则：
最小APY=K/M最大=8127yl/93569yl x 3000%=260.56%
最大APY=K/M最小=8127yl/32344yl x 3000%=753.78%
**当APY大于3000%时，仅取3000%。**
由以上得出：
**APY =K/Mx 3000%**，公式分解转换,单农场APY方便算法的通用表达式为：

### APY=本农场分发池额xYL价格/平均农田做市本金（共享值均值）x30%x365

则APY由：
YL价格，本农场分发池额，平均农田做市本金（平均本金=共享均值=农场共享值总量/农田数）三者决定APY。

静态分析：
其中YL价格越高，则M值越小，当K值不变时，则APY越大。
**YL价格与APY成正比**
其中农场分发池额越大，则K值越大，当M值不变时，则APY越大。
**农场分发池额与APY成正比**
其中平均农田做市本金越小，则M值越小，当K值不变时，则APY越大。
**平均农田做市本金与APY成反比**
当农场内有农田退出时，则场内APY越大。

动态分析：
1.最好理解的是整个生态的YL的价格因素决定了APY的大小。

2.农场分发池额由流速累积产生，而流速由共享值增量控制，即由农场激活后**再复投的本金**决定。

3.平均农田做市本金，则看总投入农田的本金多少决定，有原始田的玩家在YLE生态早期阶段APY效益比较低回本慢，而50U小田则先期回本快。
即**再复投的本金与APY成正比，农田总投入本金与APY成反比**，人话就是复投多收益大，单田投入本金多回本慢。
由于子农场流速大小受总流速（资金池大小，肥力值增量比，价格稳定比）影响，总流速受蓄水池额影响，蓄水池额受YL增发量影响，YL增发量受新建田影响，受本金投入比影响，所以**新田数量与APY成正比**。

归纳
简化通用公式：
APY=本农场分发池额xYL价格/农场平均农田做市本金（共享值均值）x30%x365
在农场内单田想要获得
直接因素：
**YL价格与APY成正比**
**农场分发池额与APY成正比**
**平均农田做市本金（共享值均值）与APY成反比**
间接因素：
**农田总投入本金与APY成反比**
**农田载种NFT与农场APY分配成正比**
**共享值增量（复投的本金）与APY成正比**
**总资金池与APY成正比**
**肥力值增量比与APY成正比**
**YL价格稳定比与APY成正比**
**新田数量与APY成正比**

**Comment by otto-001** [comment]
厉害，学习了！
这是核心，理解了这些，才能玩转YLE。掌握APY财务密码，必备知识点😁✌️

**Reply by yancanlou** [reply]
有问题的地方请指点，方向大致如此。
有兴趣的玩家可以自己建个模型或者推一下算法找找比值。
对个单农田来讲，其中NFT加成与APY成正比，NFT以最小投入额获得了单农田的成系数的增长加成，成为子农场内部的收益先锋。

**Reply by cryptosoul-vioiv** [reply]
基本田肥力加成是2倍，八宝树NFT加成后是3倍，提升了50%的收益。
以满额基本田为例，500U的做市本金，365%的APY（日收益率为1%）：
不种树：500U*2倍加成*1%=10U
种树后：500U*3倍加成*1%=15U
在不考虑复投的复投利情况下，每日收益差额为5U，八宝树成本500U，那么100天后即可收回树的成本。
在APY越高，购树成本的回本速度越快。

**Reply by linshande** [reply]
这些所谓的农田加成提升来自于子农场内部的多分配，需看准各条件高值先与人收才有，属于内部博弈的乐趣，本质上内卷性质。并不改变外部进来流速的积累！

**Reply by cryptosoul-vioiv** [reply]
这个内部博弈其实挺有意思的。
当K值增量大于M值增量量，APY是逐渐慢慢升高的，当达到某玩家预期的APY后，少部分玩家开始收取收益，此时M值减少引起APY进一步提升，于是更多的玩家收取收益，以致APY在K值减少速度小于M值减少速度下，快速达到一个峰值；之后由于K值下降速度超过M值下降速度，APY开始降低，收取收益的玩家数量逐渐减少。
收取收益的玩家，在增加共享值来提升流速，或进行新基本田做市增发YL，让K值境又大于M值增量，APY开始了新一轮的上升。
如此往复。

**Comment by cryptosoul-vioiv** [comment]
APY的大小，关键在于各农场间流速的竞争。
按目前**_“共享值增量占比决定流速占比”_**情况下，农场内玩家团结一致，形成良好的共识，就成了核心要素。

**Reply by linshande** [reply]
没错！增量也是最终要转化分发池额，而对其分配占优也是场内玩家需要的付出！
众所周知，YL价格也是非常敏感提升APY的决定性因素，这个是YLE生态整体发展要努力的。

**Comment by yancanlou** [comment]
查看本农场农田的APY，会发现显示出来的APY比通用算法还高？
因为资金池的做市的资金U下降引起YL价格下跌了，造成做市的本金额变相地减少了。
即投入做市的YL数量x实时价格下降=估值下降，这种情况下意味平均农田做市本金也减小了。
根据*平均农田做市本金与APY成反比*，是为一段下跌时间后实时本金低于最初共享值。
所以显示的APY变高了，这种假高，只是本金按现价折下后变低了。
此类为浮亏下的假高，只是参照点不同引起的。

**Comment by cryptosoul-vioiv** [comment]
还有一个细节要关注的：是否可以将所有收益，只复投到野田，提升增量共享值，无限的增大共享农场的流速？
答案是否定的！因为会受限于共享值上限。
一块Famland的收益，由原始田、基本田、野田三个地块一同产出，但只有基本田的产出，才可以提升共享值上限。

`举例：1YL=1U时，Famland里，基本田做市资金100U，野田做市资金100U；两地块的做市资金能效分别为：200U和100U，产出的收益占比为2：1`
`收取收益10YL（基本田占6.6YL），共享值上限由100提升为106.6；把所有收益全部复投到野田，共享金额度为10；`
`那么，该Famland共享值最大的可提升仍为6.6。所以，并不是并不是野田做市的所有共享金额度都可以转化为增量共享值`
`从最优资金使用效率而言，野田复投资金应为6.6，而多余的产出，则可以另外创建新Famland做市，利用新基本田里的产出来提升增量共享值。`

故，机制上：
以增量共享值进行流速的博弈，引导在野里复投；
以收益产出提升共享值上限，揭制资本大鳄在野田短期性做市快速提升增量共享值，来提升流速并套利的可能性；引导新基本田做市，对共享值扩容，同时增发YL。

![33](https://user-images.githubusercontent.com/81346056/173276810-edd70c9d-a5c6-458f-834f-1e10f62ad5ae.png)

**Comment by cryptosoul-vioiv** [comment]
肥力值转化，进行肥力加成系数的提升，也是提升收益的因素。

白皮书内容：
`Famland加持YLE生态内的CryptoTree NFT后，肥力加成可以按照0.001为单位进行提升，首次肥力加成提升0.001，肥力值须提升20000，肥力加成再次提升0.001，需提升的肥力值在之前的提升的基础值上提升3‰。`
`由此可见，Famland中，Famland的耕种时长越长、投入资金越大、肥力值越高、加持NFT后，进而转化的加成系数越多，耕种收益越高。`




![photo_2022-06-13_13-53-16](https://user-images.githubusercontent.com/81346056/173288518-93b1e428-14c4-4593-8a08-4de2feeb64ff.jpg)

**Comment by mjy368837** [comment]
你好
